### PR TITLE
Expand Centralized Ecosystem Format with Language Version Information for Bundler

### PR DESCRIPTION
**Pull request by kbukum1**
### What are you trying to accomplish?

This PR adds language version details for Bundler into the centralized ecosystem structure in Dependabot. The goal is to enhance consistency in representing package manager and language information across ecosystems.

### Anything you want to highlight for special attention from reviewers?

This update follows the existing centralized format and extends it to include language versions for Bundler, laying the groundwork for other ecosystems like Composer and npm_and_yarn.

### How will you know you've accomplished your goal?

- Bundler language version information is accessible in the centralized ecosystem structure.
- Tests verify that functionality works as expected without causing regressions.

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes, including additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: bundler/lib/dependabot/bundler/file_parser.rb [hunk 1]
```diff
@@ -327,6 +335,51 @@ def imported_ruby_files
...
+        return bundler_raw_version if defined?(@bundler_raw_version)
+
+        package_manager = PackageManager.new(bundler_version)
+
+        # If selected version is unsupported, we are going to throw unsupported error
+        # So we shoudn't try to get the raw version
+        return bundler_version if package_manager.unsupported?
+
+        # read raw version directly from the ecosystem environment
+        bundler_raw_version = SharedHelpers.in_a_temporary_repo_directory(
```
Comment (kbukum1) [inline bundler/lib/dependabot/bundler/file_parser.rb:349]: **Review Tip:** When retrieving the full raw version of Bundler, note that if an unsupported Bundler version (e.g., `"1"`) is specified, retrieving the raw version won’t be possible. In such cases, an "unsupported version" is getting raised error so we won't need to fetch the raw version.

File: bundler/lib/dependabot/bundler/file_parser.rb [hunk 2]
```diff
@@ -327,6 +335,51 @@ def imported_ruby_files
...
+          )
+        end
+        bundler_raw_version || ::Bundler::VERSION
+      end
+
+      sig { returns(String) }
+      def ruby_raw_version
+        return @ruby_raw_version if defined?(@ruby_raw_version)
+
+        ruby_raw_version = SharedHelpers.in_a_temporary_repo_directory(
```
Comment (kbukum1) [inline bundler/lib/dependabot/bundler/file_parser.rb:368]: **Review Tip**: We want to find the raw version for language directly from the ecosystem environment.

File: bundler/lib/dependabot/bundler/file_parser.rb [hunk 3]
```diff
@@ -47,7 +50,16 @@ def ecosystem
...
+        @package_manager ||= PackageManager.new(bundler_raw_version)
+      end
+
+      sig { returns(T.nilable(Ecosystem::VersionManager)) }
+      def language
+        return @language if defined?(@language)
+
+        return nil unless package_manager.unsupported?
+
+        Language.new(ruby_raw_version)
```
Comment (kbukum1) [inline bundler/lib/dependabot/bundler/file_parser.rb:62]: **Review Tip**: If the package manager is unsupported, the language version may not be retrievable. Additionally, fetching the language version is unnecessary, as the process will raise an unsupported package manager error.

**Review by honeyankit — APPROVED** [review]
LGTM

**Pull request merged**